src/components/Form.js

```diff
@@ -1,9 +1,46 @@
-const Form = () => (
-  <form className="form">
-    <input type="text" name="author" placeholder="Author" className="author" />
-    <input type="text" name="book" placeholder="Book" className="book" />
-    <input type="submit" value="Submit" className="submit" />
-  </form>
-);
+import { useSelector, useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/books';
+
+const Form = () => {
+  const books = useSelector((state) => state.books);
+  const dispatch = useDispatch();
+  return (
+    <form className="form">
+      <input
+        type="text"
+        name="title"
+        placeholder="Author"
+        className="title"
+      />
+      <input
+        type="text"
+        name="author"
+        placeholder="Book"
+        className="author"
+      />
+      <input
+        type="submit"
+        value="Submit"
+        className="submit"
+        onClick={(event) => {
+          event.preventDefault();
+          const title = event.target.previousElementSibling.value;
+          const author = event.target.previousElementSibling.previousElementSibling.value;
+          if (title && author) {
+            dispatch(addBook(books.length + 1, title, author));
+            // eslint-disable-next-line no-param-reassign
+            event.target.previousElementSibling.previousElementSibling.value = '';
+            // eslint-disable-next-line no-param-reassign
+            event.target.previousElementSibling.value = '';
+          } else {
+            // eslint-disable-next-line no-alert
+            alert('Please fill in all fields');
+          }
+        }}
+      />
+
+    </form>
+  );
+};
 
```

Comment:
- [ ] Kindly refactor your Form component to be a controlled component. This means the values of the `title` and `author` inputs should be set by the local state using the `onChange` method, not the direct inputs of the user.

- [ ] Additionally, what should be dispatched to the store is the state also, not the direct user inputs.

  You can refer to [this quick tutorial on how to set up controlled inputs](https://dmitripavlutin.com/controlled-inputs-using-react-hooks/) in React.